src/digest.rs

```diff
@@ -448,8 +447,66 @@ pub extern fn SHA512_4(out: *mut u8, out_len: c::size_t,
     polyfill::slice::fill_from_slice(out, digest);
 }
 
+unsafe extern fn sha1_block_data_order(state: *mut u64, data: *const u8, num: c::size_t) {
+    sha1_process_blocks(
+        &mut *(state as *mut _),
```

Comment:
In general we don't use `as` in *ring* as we consider `as` to be too unsafe in general. We have `polyfill::slice::u64_as_u32` for this kind of conversion.